test/example.c

```diff
@@ -19,13 +19,105 @@
 #  define TESTFILE "foo.gz"
 #endif
 
-#define CHECK_ERR(err, msg) { \
-    if (err != Z_OK) { \
-        fprintf(stderr, "%s error: %d\n", msg, err); \
-        exit(1); \
+#define SUCCESSFUL                 1
+#define FAILED_WITH_ERROR_CODE     0
+#define FAILED_WITHOUT_ERROR_CODE -1
+
+typedef struct test_result_s {
+    int         result; /* One of: SUCCESSFUL,
+                           FAILED_WITH_ERROR_CODE, or
+                           FAILED_WITHOUT_ERROR_CODE*/
+    int         error_code; /* error code if success is FAILED_WITH_ERROR_CODE */
+    int         line_number;
+    const char* message;
+    const char* extended_message;
+} test_result;
+
+#define STRING_BUFFER_SIZE 100
+char string_buffer[STRING_BUFFER_SIZE];
+
+#define CHECK_ERR(_error_code, _message) { \
+    if (_error_code != Z_OK) { \
+        test_result result; \
+        result.result = FAILED_WITH_ERROR_CODE; \
+        result.error_code = _error_code; \
+        result.line_number = __LINE__; \
+        result.message = _message; \
+        return result; \
```

Comment:
extended_message uninitialized, unpredictable results.

Reply:
The handler for FAILED_WITH_ERROR_CODE doesn't try to access the extended message. See line 80.

Reply:
Next guy might reference it.  *cough*

Reply:
Agreed. It is like a time bomb / trap for the next person to fall into.
I'll initialize all the variables.